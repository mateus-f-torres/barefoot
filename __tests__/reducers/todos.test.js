import todosReducer from '../../src/reducers/todos.js';

describe('Todos Reducer', () => {
  const defaultState = [
    {
      id: 0,
      text: 'Buy bread',
      completed: false,
    },
    {
      id: 1,
      text: 'Take dog for a walk',
      completed: true,
    },
    {
      id: 2,
      text: 'Comment codebase',
      completed: false,
    },
  ];

  describe('Unit Test', () => {
    describe('DEFAULT_STATE', () => {
      test('is correct', () => {
        const action = {type: 'dummy_action'};

        expect(todosReducer(undefined, action)).toEqual(defaultState);
      });
    });

    describe('ADD_TODO', () => {
      test('return correct state', () => {
        const action = {type: 'ADD_TODO', payload: 'test reducer'};
        const expectedState = [
          ...defaultState,
          {id: 3, text: 'test reducer', completed: false},
        ];

        expect(todosReducer(undefined, action)).toEqual(expectedState);
      });
    });

    describe('TOGGLE_TODO', () => {
      test('return correct state', () => {
        const action = {type: 'TOGGLE_TODO', payload: 2};
        const expectedItem = {
          id: 2, text: 'Comment codebase', completed: true,
        };

        expect(todosReducer(undefined, action)[2]).toEqual(expectedItem);
      });
    });

    describe('REMOVE_TODO', () => {
      test('return correct state', () => {
        const action = {type: 'REMOVE_TODO', payload: 2};

        expect(todosReducer(undefined, action).length).toEqual(2);
      });
    });
  });

  describe('Snapshot Test', () => {
    describe('INITIAL STATE', () => {
      test('matches current production snapshot', () => {
        const action = {type: 'dummy_action'};

        expect(todosReducer(undefined, action)).toMatchSnapshot();
      });
    });

    describe('ADD_TODO', () => {
      test('matches current production snapshot', () => {
        const action = {type: 'ADD_TODO', payload: 'snapshot test reducer'};

        expect(todosReducer(undefined, action)).toMatchSnapshot();
      });
    });

    describe('TOGGLE_TODO', () => {
      test('matches current production snapshot', () => {
        const action = {type: 'TOGGLE_TODO', payload: 2};

        expect(todosReducer(undefined, action)).toMatchSnapshot();
      });
    });

    describe('REMOVE_TODO', () => {
      test('matches current production snapshot', () => {
        const action = {type: 'REMOVE_TODO', payload: 2};

        expect(todosReducer(undefined, action)).toMatchSnapshot();
      });
    });
  });
});
