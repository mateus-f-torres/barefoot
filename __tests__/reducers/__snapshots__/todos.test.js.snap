// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todos Reducer Snapshot Test ADD_TODO matches current production snapshot 1`] = `
Array [
  Object {
    "completed": false,
    "id": 0,
    "text": "Buy bread",
  },
  Object {
    "completed": true,
    "id": 1,
    "text": "Take dog for a walk",
  },
  Object {
    "completed": false,
    "id": 2,
    "text": "Comment codebase",
  },
  Object {
    "completed": false,
    "id": 4,
    "text": "snapshot test reducer",
  },
]
`;

exports[`Todos Reducer Snapshot Test INITIAL STATE matches current production snapshot 1`] = `
Array [
  Object {
    "completed": false,
    "id": 0,
    "text": "Buy bread",
  },
  Object {
    "completed": true,
    "id": 1,
    "text": "Take dog for a walk",
  },
  Object {
    "completed": false,
    "id": 2,
    "text": "Comment codebase",
  },
]
`;

exports[`Todos Reducer Snapshot Test REMOVE_TODO matches current production snapshot 1`] = `
Array [
  Object {
    "completed": false,
    "id": 0,
    "text": "Buy bread",
  },
  Object {
    "completed": true,
    "id": 1,
    "text": "Take dog for a walk",
  },
]
`;

exports[`Todos Reducer Snapshot Test TOGGLE_TODO matches current production snapshot 1`] = `
Array [
  Object {
    "completed": false,
    "id": 0,
    "text": "Buy bread",
  },
  Object {
    "completed": true,
    "id": 1,
    "text": "Take dog for a walk",
  },
  Object {
    "completed": true,
    "id": 2,
    "text": "Comment codebase",
  },
]
`;
